Remove react scripts ts

jpreese opened the pull request:
Conversion from react-scripts-ts to react-scripts.

This should resolve #1

Commit:
Remove react-scripts-ts

Commit:
Add react-scripts

Commit:
Adding supporting packages

Commit:
Remove obsolete files

Commit:
Update npm scripts

Commit:
Add default browser

Commit:
Add default files

Commit:
Explicitly use enzyme and Adapter

numo16 commented:
Looks like your package-lock.json is referencing a private repository, following the removing and adding of various packages. We could resolve this by having a [project level .npmrc file](https://docs.npmjs.com/files/npmrc#per-project-config-file) that explicitly uses `registry="https://registry.npmjs.org/"`.

Additionally, related to the topic of package management, we should probably not be mixing yarn and npm for package management in this repo. Standardizing on just npm and ditching the yarn.lock file in this repo probably makes the most sense.

Commit:
Add npmrc file

Commit:
Update package-lock to use new registry

jpreese commented:
Good catch! Added npmrc to prevent this from occurring in the future and updated the package-lock.

I would personally prefer to only use npm. I'll open an issue.

numo16 commented on .npmrc:
```diff
@@ -0,0 +1 @@
+registry="https://registry.npmjs.org/"
```
ignore previous comment, this should be fine

numo16 commented:
I'm encountering something odd after these changes. When running the app locally, in chrome 72.0.3626.96 or firefox 65.0.1, the definitions editor seems to be rendering incorrectly. The contents of the editor are blank (no line numbers, no default text, no scroll bars). Here is a screenshot of what I'm seeing in both browsers:

![image](https://user-images.githubusercontent.com/268388/52816445-3f1ed880-306f-11e9-9021-57f037d01371.png)

I checked in react devtools and the value is definitely being set on the component, so it could be that whatever happened in switching over to react-scripts is changing how the css is being bundled?

![image](https://user-images.githubusercontent.com/268388/52816572-8efd9f80-306f-11e9-8ea3-b48404ca77f6.png)

jpreese commented:
Similar to https://github.com/graphql/graphiql/issues/33 , this does seem to be a loading issue. If I call codemirror.css directly from public/index.html, rather than an import, the issue does go away.

numo16 commented:
Noticing that vendoring in public fixes it. It is also called out in [create react apps's](https://facebook.github.io/create-react-app/docs/using-the-public-folder) docs as an escape hatch for things that need to be loaded explicitly. The downside to doing it that way is that we have to be conscious of that stylesheet being vendored in the public folder if we ever upgrade the codemirror package.

jpreese commented:
Yeah, I moved codemirror from node_modules to the public dir to force it to load ahead of the app, but it felt a little wrong for the reason you mentioned. Though it's currently the only solution I've found.

Commit:
Move codemirror stylesheet to public directory

jpreese commented:
Moved the codemirror stylesheet to the public directory

numo16 commented:
Everything is working as expected now and takes care of the referenced issue. Update the changelog (since dependency loading was changed a bit) and we're good to go.

numo16 commented:
Regarding versioning in the changelog, semantic versioning doesn't tend to fit well with applications. It would be best to move to just updating the changelog with releases by date.

Commit:
Update CHANGELOG

numo16 reviewed (approved):
Everything looks good.
